src/bosh_azure_cpi/lib/cloud/azure/cloud.rb

```diff
@@ -479,6 +479,44 @@ def resize_disk(disk_cid, new_size)
       end
     end
 
+    def update_disk(disk_cid, new_size, cloud_properties)
+      @logger.info("update_disk(#{disk_cid}, #{new_size}, #{cloud_properties})")
+      raise Bosh::Clouds::NotSupported, 'Native disk update only supported for managed disks' unless @use_managed_disks
+
+      @telemetry_manager.monitor('initialize') do
+        _init_azure
+      end
+
+      with_thread_name("update_disk(#{disk_cid}, #{new_size}, #{cloud_properties})") do
+        @telemetry_manager.monitor('update_disk', id: disk_cid) do
+          disk_id = DiskId.parse(disk_cid, _azure_config.resource_group_name)
+          disk_name = disk_id.disk_name
+          disk = @disk_manager2.get_data_disk(disk_id)
+          raise Bosh::Clouds::NotSupported, "Disk '#{disk_name}' not found" if disk.nil?
+
+          caching = cloud_properties['caching']
+          raise Bosh::Clouds::NotSupported, 'Disk caching cannot be modified' if caching && (caching != disk[:caching])
+
+          new_size_in_gib = mib_to_gib(new_size)
+          old_size_in_gib = disk[:disk_size]
+          raise Bosh::Clouds::NotSupported, 'Disk size cannot be decreased' if old_size_in_gib > new_size_in_gib
+
+          new_size_in_gib = nil if new_size_in_gib == old_size_in_gib
+          account_type = cloud_properties['storage_account_type']
```

Comment:
@s4heid do you want to implement the snapshot based migration if the storage type is chaged from Premium V2/UltraSSD -> Premium v1/Standard SSD/Standard HDD?

Reply:
Yes, I think the snapshot-based storage type change is a useful feature. I have planned to do some testing next week and eventually provide a pull request if everything goes well.

Reply:
I don't see it as a requirement for this change, though. The Azure API returns a meaningful error when trying to switch from PremiumV2 to a lower tier storage account.